app/assets/src/components/views/discovery/DiscoveryView.jsx

```diff
@@ -106,41 +102,58 @@ class DiscoveryView extends React.Component {
             : "projects",
         filteredProjectDimensions: [],
         filteredSampleDimensions: [],
+        filteredProjectStats: {},
         filteredSampleStats: {},
         filters: {},
         loadingDimensions: true,
         loadingLocations: true,
-        loadingProjects: true,
         loadingStats: true,
-        loadingVisualizations: true,
         mapLevel: "country",
         mapLocationData: {},
         mapPreviewedLocationId: null,
-        mapPreviewedProjects: [],
-        mapPreviewedSampleIds: [],
-        mapPreviewedSamples: [],
         mapSidebarProjectDimensions: [],
         mapSidebarSampleDimensions: [],
         mapSidebarSampleStats: {},
         mapSidebarTab: "summary",
         project: null,
         projectDimensions: [],
         projectId: projectId,
-        projects: [],
         rawMapLocationData: {},
         sampleActiveColumns: undefined,
         sampleDimensions: [],
         search: null,
         selectedSampleIds: new Set(),
         showFilters: true,
         showStats: true,
-        visualizations: [],
       },
       localState,
       sessionState,
       urlState
     );
     this.dataLayer = new DiscoveryDataLayer(this.props.domain);
+    const conditions = this.getConditions();
+    this.samples = this.dataLayer.samples.createView({ conditions });
+    this.projects = this.dataLayer.projects.createView({
+      conditions,
+      onViewChange: this.refreshProjectStats,
+    });
+    this.visualizations = this.dataLayer.visualizations.createView({
+      conditions,
+    });
+    this.mapPreviewProjects = this.projects;
+    this.mapPreviewSamples = this.samples;
+
+    // hold references to the views to allow resetting the tables
+    this.projectsView = null;
```

Comment:
I'm not sure if this is even in this PR, but when I click into a project and then click back, it seems filtered to just that project still: 
![Sep-19-2019 17-52-48](https://user-images.githubusercontent.com/5652739/65291013-745c3f00-db06-11e9-957b-f5b83ddd5ed4.gif)

Reply:
Fixed.